internal/integration/cli/config.go

```diff
@@ -61,6 +62,25 @@ func (suite *TalosconfigSuite) TestMerge() {
 	suite.Require().NotNil(c.Contexts["foo-1"])
 }
 
+// TestNew checks `talosctl config new`.
+func (suite *TalosconfigSuite) TestNew() {
+	tempDir := suite.T().TempDir()
+
+	node := suite.RandomDiscoveredNode(machine.TypeControlPlane)
+
+	talosconfigPath := filepath.Join(tempDir, "talosconfig")
+	suite.RunCLI([]string{"--nodes", node, "config", "new", "--roles", "os:reader", talosconfigPath},
+		base.StdoutEmpty())
+
+	suite.RunCLI([]string{"--nodes", node, "--talosconfig", talosconfigPath, "ls", "/etc/hosts"},
+		base.StdoutShouldMatch(regexp.MustCompile(`hosts`)))
+
+	suite.RunCLI([]string{"--nodes", node, "--talosconfig", talosconfigPath, "read", "/etc/hosts"},
```

Comment:
can we add more negative test cases here to make sure that `reader` can't break out?

can't read machine config, secrets, can't execute kubeconfig, reset, etc.?